File: sdkwork-ai-sdk-go/api/file.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type FileApi struct {
    client *sdkhttp.Client
}

func NewFileApi(client *sdkhttp.Client) *FileApi {
    return &FileApi{client: client}
}

// List files
func (a *FileApi) ListFiles(query map[string]interface{}) (sdktypes.OpenAiFileList, error) {
    raw, err := a.client.Get(AiApiPath("/files"), query, nil)
    if err != nil {
        var zero sdktypes.OpenAiFileList
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiFileList](raw)
}

// Upload file
func (a *FileApi) Upload(body *sdktypes.UploadFileRequest, query map[string]interface{}) (sdktypes.OpenAiFileObject, error) {
    raw, err := a.client.Post(AiApiPath("/files"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.OpenAiFileObject
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiFileObject](raw)
}

// Retrieve file content
func (a *FileApi) RetrieveFileContent(file_id string) (string, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/files/%s/content", file_id)), nil, nil)
    if err != nil {
        var zero string
        return zero, err
    }
    return decodeResult[string](raw)
}

// Retrieve file
func (a *FileApi) Retrieve(file_id string) (sdktypes.OpenAiFileObject, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/files/%s", file_id)), nil, nil)
    if err != nil {
        var zero sdktypes.OpenAiFileObject
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiFileObject](raw)
}

// Delete file
func (a *FileApi) DeleteFile(file_id string) (sdktypes.OpenAiFileDeleteResponse, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/files/%s", file_id)), nil, nil)
    if err != nil {
        var zero sdktypes.OpenAiFileDeleteResponse
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiFileDeleteResponse](raw)
}

File: sdkwork-ai-sdk-typescript/src/api/file.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { OpenAiFileDeleteResponse, OpenAiFileList, OpenAiFileObject } from '../types';


export class FileApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** List files */
  async listFiles(params?: QueryParams): Promise<OpenAiFileList> {
    return this.client.get<OpenAiFileList>(aiApiPath(`/files`), params);
  }

/** Upload file */
  async upload(body?: FormData, params?: QueryParams): Promise<OpenAiFileObject> {
    return this.client.post<OpenAiFileObject>(aiApiPath(`/files`), body, params);
  }

/** Retrieve file content */
  async retrieveFileContent(file_id: string | number): Promise<string> {
    return this.client.get<string>(aiApiPath(`/files/${file_id}/content`));
  }

/** Retrieve file */
  async retrieve(file_id: string | number): Promise<OpenAiFileObject> {
    return this.client.get<OpenAiFileObject>(aiApiPath(`/files/${file_id}`));
  }

/** Delete file */
  async deleteFile(file_id: string | number): Promise<OpenAiFileDeleteResponse> {
    return this.client.delete<OpenAiFileDeleteResponse>(aiApiPath(`/files/${file_id}`));
  }
}

export function createFileApi(client: HttpClient): FileApi {
  return new FileApi(client);
}
